Zoom control

sspanak opened the pull request:
Hi, Ben. These are the zoom related changes I wrote you about this week.

Please, review and comment, or merge, if it looks okay.

bkrem commented on src/Tree/index.js:
```diff
@@ -82,6 +82,13 @@ export default class Tree extends React.Component {
           .scaleExtent([scaleExtent.min, scaleExtent.max])
           .on('zoom', () => {
             g.attr('transform', `translate(${event.translate}) scale(${event.scale})`);
+            if (typeof onUpdate !== 'undefined') {
```
Unless there's a specific reason for the difference I'd say let's keep this consistent with line 198; i.e. also make it `typeof onUpdate === 'function'`

sspanak replied:
Fixed. This is a matter avoiding an error, not just consistency. 😄

bkrem commented:
Hey @sspanak,

Great work on this! Minor comment about consistent typechecking but otherwise I think we're almost good to go 🎉 

Could you write out a brief description for each new top-level prop? E.g. `zoom - Sets an initial zoom level` or similar :) 
Then I'll go ahead and do the docs before releasing this.

bkrem commented:
Also let's revert the version number commit since this is actually a minor not a patch release (i.e. we're going to `1.9.0` + the release script will take care of that anyways ;)

sspanak commented:
Actually, hold on. I am calling "onUpdate()" when a node is clicked, but this is incorrect. Right before that state is being set, which causes a re-render, which is the actual component update. I need to fix this.

bkrem commented:
Thanks for the amendments and double-checking your work! Haven't had time to check over the new commits yet but will do so tonight.

sspanak commented:
Yes, please do so. I do not expect to do any other changes. I suppose it will be easier to view all changes at once, instead of going commit by commit, but it is up to you.

bkrem commented on src/Tree/index.js:
```diff
@@ -39,10 +57,14 @@ export default class Tree extends React.Component {
       });
     }
 
+    this.internalState.d3.translate = this.props.translate;
```
@sspanak

bkrem commented on src/Tree/index.js:
```diff
@@ -27,8 +34,19 @@ export default class Tree extends React.Component {
     this.bindZoomListener(this.props);
-    // TODO find better way of setting initialDepth, re-render here is suboptimal
-    this.setState({ initialRender: false }); // eslint-disable-line
+    this.internalState.initialRender = false;
```
Nice! :) That solution seems so obvious now that I see it.

bkrem commented:
@sspanak Looking great, just those two assignments in `componentWillReceiveProps` to correct and then we're good to go 🚀

sspanak commented:
I have removed the D3 state overwriting, because it is not necessary. Tree position or scale do not change when new props are received. 

Actually, is this the intended behaviour? I have just introduced zoom property, but how about translation? I thought I might have broken it, but it is the same in master.

bkrem commented:
Great, thanks again for your awesome work on this! Will merge now and publish with a small fix related to #77 later today.

From what you're describing this sounds like expected behaviour, but if you think that behaviour has a good reason to be different by default feel free to open an issue to discuss :)

sspanak commented:
Sorry, my bad. I was testing with the same values "before" and "after". It's all good, there is no bug. :)

bkrem commented:
Released in [1.9.1](https://github.com/bkrem/react-d3-tree/releases/tag/v1.9.1) 🎉
